django/chat/utils.py

```diff
@@ -203,14 +208,29 @@ def sse_string(message: str, format=True, dots=False, remove_stop=False) -> str:
             response_replacer = stream_to_replacer([response_str])
 
         # Stream the response text
+        first_message = True
         async for response in response_replacer:
+            if first_message and switch_mode:
+                full_message = render_to_string(
+                    "chat/components/mode_switch_message.html",
+                    {
+                        "mode": mode,
+                        "mode_str": mode_str,
+                        "library_id": chat.options.qa_library_id,
+                        "library_str": chat.options.qa_library.name,
+                    },
+                )
+                yield sse_string(full_message, format=False, dots=dots_html)
+                await asyncio.sleep(1)
```

Comment:
1 second delay after switching modes to display the "Switched modes" message to the user.